src/corppa/poetry_detection/annotation/process_adjudication_data.py

```diff
@@ -0,0 +1,176 @@
+import argparse
+import csv
+import pathlib
+import sys
+from collections.abc import Generator
+from typing import Any
+
+import orjsonl
+from tqdm import tqdm
+from xopen import xopen
+
+
+def get_excerpts(page_annotation: dict[str, Any]) -> list[dict[str, int | str]]:
+    """
+    Extract excerpts from page-level annotation. Excerpts have the following
+    fields:
+        * start: character-level starting index
+        * end: character-level end index (Pythonic, exclusive)
+        * text: text of page excerpt
+
+    Note: Currently ignoring span labels, since there's only one for the
+          poetry detection task.
+    """
+    excerpts = []
+    # Blank pages may not have a text field, so in these cases set to empty string
+    page_text = page_annotation.get("text", "")
+    if "spans" not in page_annotation:
+        raise ValueError("Page annotation missing 'spans' field")
+    for span in page_annotation["spans"]:
+        excerpt = {
+            "start": span["start"],
+            "end": span["end"],
+            "text": page_text[span["start"] : span["end"]],
+        }
+        excerpts.append(excerpt)
+    return excerpts
+
+
+def process_page_annotation(page_annotation) -> dict[str, Any]:
+    """
+    Extracts desired content from page-level annotation. The returned data has
+    the following fields"
+        * page_id: Page's PPA page identifier
+        * work_id: PPA work identifier
+        * work_title: Title of PPA work
+        * work_author: Author of PPA work
+        * work_year: Publication of PPA work
+        * n_excerpts: Number of poetry excerpts contained in page
+        * excerpts: List of poetry excerpts identified within page
+    """
+    page_data = {}
+    page_data["page_id"] = page_annotation["id"]
+    page_data["work_id"] = page_annotation["work_id"]
+    page_data["work_title"] = page_annotation["meta"]["title"]
+    page_data["work_author"] = page_annotation["meta"]["author"]
+    page_data["work_year"] = page_annotation["meta"]["year"]
+    page_data["excerpts"] = get_excerpts(page_annotation)
+    page_data["n_excerpts"] = len(page_data["excerpts"])
+    return page_data
+
+
+def get_excerpt_entries(page_data: dict[str, Any]) -> Generator[dict[str, Any]]:
+    """
+    Generate excerpt entries data from the processed page produced by
+    `process_page_annotation`.
+    """
+    for excerpt in page_data["excerpts"]:
+        entry = {
+            "page_id": page_data["page_id"],
+            "work_id": page_data["work_id"],
+            "work_title": page_data["work_title"],
+            "work_author": page_data["work_author"],
+            "work_year": page_data["work_year"],
+            "start": excerpt["start"],
+            "end": excerpt["end"],
+            "text": excerpt["text"],
```

Comment:
when/if we come back to this (if we use this script again), let's tweak these fields to match the ones we've agreed on for the poem dataset